doc/examples/streamline_registration.py

```diff
@@ -0,0 +1,322 @@
+
+identity = np.eye(4)
+affine_map = AffineMap(identity, static.shape, static_affine, moving.shape,
+                       moving_affine)
+resampled = affine_map.transform(moving)
+
+"""
+We specify the mismatch metric:
+
+"""
+
+nbins = 32
+sampling_prop = None
+metric = MutualInformationMetric(nbins, sampling_prop)
+
+"""
+As well as the optimization strategy:
+
+"""
+
+level_iters = [10, 10, 5]
+sigmas = [3.0, 1.0, 0.0]
+factors = [4, 2, 1]
+affine_reg = AffineRegistration(metric=metric, level_iters=level_iters,
+                                sigmas=sigmas, factors=factors)
+transform = TranslationTransform3D()
+
+params0 = None
+translation = affine_reg.optimize(static, moving, transform, params0,
+                                  static_affine, moving_affine)
+transformed = translation.transform(moving)
+transform = RigidTransform3D()
+
+rigid_map = affine_reg.optimize(static, moving, transform, params0,
+                                static_affine, moving_affine,
+                                starting_affine=translation.affine)
+transformed = rigid_map.transform(moving)
+transform = AffineTransform3D()
+
+"""
+We bump up the iterations to get a more exact fit:
+
+"""
+
+affine_reg.level_iters = [1000, 1000, 100]
+affine_map = affine_reg.optimize(static, moving, transform, params0,
+                                 static_affine, moving_affine,
+                                 starting_affine=rigid_map.affine)
+transformed = affine_map.transform(moving)
+
+
+"""
+We now perform the non-rigid deformation using the Symmetric Diffeomorphic
+Registration (SyN) Algorithm proposed by Avants et al. [Avants09]_ (also
+implemented in the ANTs software [Avants11]_):
+
+"""
+
+from dipy.align.imwarp import SymmetricDiffeomorphicRegistration
+from dipy.align.metrics import CCMetric
+
+metric = CCMetric(3)
+level_iters = [10, 10, 5]
+sdr = SymmetricDiffeomorphicRegistration(metric, level_iters)
+
+mapping = sdr.optimize(static, moving, static_affine, moving_affine,
+                       affine_map.affine)
+warped_moving = mapping.transform(moving)
+warped_static = mapping.transform_inverse(moving)
+
+"""
+We show the registration result with:
+
+"""
+from dipy.viz import regtools
+
+regtools.overlay_slices(static, warped_moving, None, 0, "Static", "Moving",
+                        "transformed_sagittal.png")
+regtools.overlay_slices(static, warped_moving, None, 1, "Static", "Moving",
+                        "transformed_coronal.png")
+regtools.overlay_slices(static, warped_moving, None, 2, "Static", "Moving",
+                        "transformed_axial.png")
+
+"""
+.. figure:: transformed_sagittal.png
+   :align: center
+.. figure:: transformed_coronal.png
+   :align: center
+.. figure:: transformed_axial.png
+   :align: center
+
+   Deformable registration result.
+"""
+
+"""
+We read the streamlines from file in voxel space:
+
+"""
+
+from dipy.io.streamline import load_trk
+
+streamlines, hdr = load_trk('lr-superiorfrontal.trk')
```

Comment:
`load_trk` is changing and need an affine now. I advise you to rebase this PR

Reply:
agreed. @jhlegarreta -- do you want to handle the rebase?

Reply:
f66eb9c was rebased on `master`.  Using the new `load_tractogram` method (I thought we would be willing to eventually move all legacy methods to this common interface, so this is one less).